test/runner/run.js

```diff
@@ -67,15 +63,47 @@ export async function run( {
 		switch ( message.type ) {
 			case "testEnd":
 				touchWorker( message.id );
-				reportTest( message.data, reports[ message.id ] );
+				const errors = reportTest(
+					message.data,
+					message.id,
+					reports[ message.id ]
+				);
+				if ( errors ) {
+					pendingErrors[ message.id ] ||= {};
+					pendingErrors[ message.id ].errors = errors;
+				}
 				break;
 			case "runEnd": {
-				failedCount += reportEnd( message.data, reports[ message.id ] );
-				await cleanupWorker( message.id, verbose );
+				const failedCount = reportEnd(
+					message.data,
+					message.id,
+					reports[ message.id ]
+				);
+
+				if ( !failedCount ) {
+					if ( pendingErrors[ message.id ] ) {
```

Comment:
What are pending errors? How they are used to detect flaky tests?

Reply:
Pending errors are cleared if a retry passes.